tests/functional/test_new_resolver.py

```diff
@@ -189,7 +189,42 @@ def test_new_resolver_ignore_dependencies(script):
     assert_not_installed(script, "dep")
 
 
-def test_new_resolver_installs_extras(script):
+@pytest.mark.parametrize(
+    "root_dep",
+    [
+        "base[add]",
+        "base[add] >= 0.1.0",
+        pytest.param(  # Non-standard syntax. To deprecate, see pypa/pip#8288.
+            "base >= 0.1.0[add]",
+            marks=pytest.mark.skipif(
+                "sys.platform == 'win32'",
+                reason="script.pip() does not handle >= on Windows",
+            ),
+        ),
+    ],
+)
+def test_new_resolver_installs_extras(script, root_dep):
+    create_basic_wheel_for_package(
+        script,
+        "base",
+        "0.1.0",
+        extras={"add": ["dep"]},
+    )
+    create_basic_wheel_for_package(
+        script,
+        "dep",
+        "0.1.0",
+    )
+    script.pip(
+        "install", "--unstable-feature=resolver",
+        "--no-cache-dir", "--no-index",
+        "--find-links", script.scratch_path,
+        root_dep,
```

Comment:
Why not put `root_dep` in a requirements file and supply that to pip? Avoids the command line parsing problem...

Reply:
Do those two share the same parsing implementations? I heard there are some subtle differences and am hesitant to do this.

Reply:
TBH, I never even thought about that possibility. Let me try it, one sec.

Reply:
```python
def test_new_resolver_installs_extras(script, root_dep):
    create_basic_wheel_for_package(
        script,
        "base",
        "0.1.0",
        extras={"add": ["dep"]},
    )
    create_basic_wheel_for_package(
        script,
        "dep",
        "0.1.0",
    )
    requirements_file = script.scratch_path / "requirements.txt"
    requirements_file.write_text(root_dep)
    script.pip(
        "install", "--unstable-feature=resolver",
        "--no-cache-dir", "--no-index",
        "--find-links", script.scratch_path,
        "-r", requirements_file,
    )
    assert_installed(script, base="0.1.0", dep="0.1.0")
```

fails with the old code, works with the new. That's good enough for me 🙂

Reply:
I pushed the change anyway since it *seems* to work the same in this case. Someone in the future can `git blame`, find this PR, and file more issues if this becomes a problem…